kratos/utilities/auxiliar_model_part_utilities.h

```diff
@@ -412,25 +439,32 @@ class KRATOS_API(KRATOS_CORE) AuxiliarModelPartUtilities
         {
         case (DataLocation::NodeHistorical):{
             ImportDataSizeCheck(mrModelPart.NumberOfNodes(), rData.size());
-            IndexType counter = 0;
-            for(auto& r_node : mrModelPart.Nodes()){
-                auto& r_val = r_node.FastGetSolutionStepValue(rVariable);
-                r_val = rData[counter++];
-            }
+
+            ModelPart::NodesContainerType::iterator inodebegin = mrModelPart.NodesBegin();
+            IndexPartition<IndexType>(mrModelPart.NumberOfNodes()).for_each([&](IndexType Index){
+                ModelPart::NodesContainerType::iterator inode = inodebegin + Index;
```

Comment:
```suggestion
            auto inodebegin = mrModelPart.NodesBegin();
            IndexPartition<IndexType>(mrModelPart.NumberOfNodes()).for_each([&](IndexType Index){
                auto inode = inodebegin + Index;
```

here `auto` makes the code more readable IMO

Reply:
True!